wandb/sdk/launch/builder/templates/_wandb_bootstrap.py

```diff
@@ -78,16 +87,28 @@ def main() -> None:
                     # can't pip install wandb==0.*.*.dev1 through pip. Lets just install wandb for now
                     if req.startswith("wandb==") and "dev1" in req:
                         req = "wandb"
-                    reqs.append(req.strip().replace(" ", ""))
+                    match = re.match(
+                        r"torch(vision|audio)?==\d+\.\d+\.\d+(\+(?:cu[\d]{2,3})|(?:cpu))?",
+                        req,
+                    )
+                    if match:
+                        variant = match.group(2)
+                        if variant:
+                            extra_index = (
+                                f"https://download.pytorch.org/whl/{variant[1:]}"
+                            )
+                        torch_reqs.append(req.strip().replace(" ", ""))
+                    else:
+                        reqs.append(req.strip().replace(" ", ""))
                 else:
                     print(f"Ignoring requirement: {req} from frozen requirements")
                 if len(reqs) >= CORES:
-                    deps_failed = install_deps(reqs)
+                    deps_failed = install_deps(reqs, opts=OPTS)
```

Comment:
We want to use the `no-deps` option for all the other dependencies so that we don't install and uninstall a bunch of different versions of the same package as we go through the freeze, but when we install torch we don't want to use that argument. So I gave the install a kwarg for the options that should be used 🤷🏼 This seemed like the way to do it that was least prone to creating mypy issues or messing with python `global` stuff